src/main/java/com/srans/nestserver/controller/PaymentController.java

```diff
@@ -39,56 +41,133 @@
-		
-		List<Payment> payments = null;
-		logger.info("IN::GET::/payments::getPaymentHistory::" + userId+"::"+role);
-		if(role != null && role.equalsIgnoreCase(NSConstants.ROLE_TENANT)) {
-			payments = paymentRepository.getPaymentHistory(userId);
-		}else {
-			payments = paymentRepository.getAllUserPaymentsForAdmin(userId);
+
+	@Autowired
+	private RoomRepository roomRepository;
+
+	@GetMapping("/payments")
+	@PreAuthorize("permitAll()")
+	public List<PaymentHistory> getPaymentHistory(@RequestParam(value = "id") Long userId,
+			@RequestParam(value = "role") String role) throws ResourceNotFoundException {
+
+		// List<Payment> payments = null;
+		logger.info("IN::GET::/payments::getPaymentHistory::" + userId + "::" + role);
+		List<PaymentHistory> paymentList = new ArrayList<>();
+
+		PaymentHistory payment = new PaymentHistory();
+		if (role != null && role.equalsIgnoreCase(NSConstants.ROLE_TENANT)) {
+
+			paymentRepository.getPaymentHistory(userId).stream().forEach(payments -> {
+
+				if (payments.getUserId() == userId) {
+					payment.setId(payments.getId());
+					payment.setRoomRent(payments.getRoomRent());
+					payment.setBankName(payments.getBankName());
+					payment.setTransactionId(payments.getTransactionId());
+					payment.setPaymentThrough(payments.getPaymentThrough());
+					payment.setDepositAmount(payments.getDepositAmount());
+					payment.setDiscountAmount(payments.getDiscountAmount());
+					payment.setAmountToBePaid(payments.getAmountToBePaid());
+
+				}
+			});
+
+			List<Object[]> response = paymentRepository.getTenantInfo(userId);
+			// List<PaymentHistory> paymentHistory=new ArrayList<PaymentHistory>();
+
+			Long roomId = 0L;
+			for (Object[] p : response) {
+
+				roomId = (Long) p[0];
+				payment.setName((String) p[1]);
+
+			}
+
+			Room room = roomRepository.getOne(roomId);
+			payment.setRoomName(room.getRoomName());
+			payment.setRoomType(room.getRoomType());
+			paymentList.add(payment);
+
+		} else {
+
+			List<PaymentHistory> ll = new ArrayList<>();
+			paymentRepository.getAllUserPaymentsForAdmin(userId).stream().forEach(payments -> {
+				PaymentHistory paymentHistory = new PaymentHistory();
+				paymentHistory.setId(payments.getId());
+				paymentHistory.setRoomRent(payments.getRoomRent());
+				paymentHistory.setBankName(payments.getBankName());
+				paymentHistory.setTransactionId(payments.getTransactionId());
+				paymentHistory.setPaymentThrough(payments.getPaymentThrough());
+				paymentHistory.setDepositAmount(payments.getDepositAmount());
+				paymentHistory.setDiscountAmount(payments.getDiscountAmount());
+				paymentHistory.setAmountToBePaid(payments.getAmountToBePaid());
+
+				List<Object[]> response = paymentRepository.getTenantInfo(payments.getUserId());
+
+				Long roomId = 0L;
+
+				for (Object[] p : response) {
+
+					roomId = (Long) p[0];
+					System.out.println(p[1]);
+					paymentHistory.setName((String) p[1]);
+
+				}
+				Room room = roomRepository.getOne(roomId);
```

Comment:
Looks like some of and else as duplicate code, check it once, if that's ok then fine